ext/opcache/jit/zend_jit_vm_helpers.c

```diff
@@ -512,16 +512,27 @@ static int zend_jit_trace_record_fake_init_call_ex(zend_execute_data *call, zend
 		 && (func->op_array.fn_flags & (ZEND_ACC_CLOSURE|ZEND_ACC_FAKE_CLOSURE))) {
 			return -1;
 		}
-		if (func->type == ZEND_USER_FUNCTION
-		 && (func->op_array.fn_flags & ZEND_ACC_CLOSURE)) {
+		if (func->type == ZEND_USER_FUNCTION) {
 			jit_extension =
 				(zend_jit_op_array_trace_extension*)ZEND_FUNC_INFO(&func->op_array);
-			if (UNEXPECTED(!jit_extension
-			 || !(jit_extension->func_info.flags & ZEND_FUNC_JIT_ON_HOT_TRACE)
-			 || (func->op_array.fn_flags & ZEND_ACC_FAKE_CLOSURE))) {
+            if (func->op_array.fn_flags & ZEND_ACC_CLOSURE) {
+				if (UNEXPECTED(!jit_extension
+				 || !(jit_extension->func_info.flags & ZEND_FUNC_JIT_ON_HOT_TRACE)
+				 || (func->op_array.fn_flags & ZEND_ACC_FAKE_CLOSURE))) {
+					return -1;
+				}
+				func = (zend_function*)jit_extension->op_array;
+			}
+			// First not-skipped op
+			zend_op *opline = func->op_array.opcodes;
+			if (!(func->op_array.fn_flags & ZEND_ACC_HAS_TYPE_HINTS)) {
+				while (opline->opcode == ZEND_RECV || opline->opcode == ZEND_RECV_INIT) {
+					opline++;
+				}
+			}
+			if (jit_extension && ZEND_OP_TRACE_INFO(opline, jit_extension->offset)->trace_flags & ZEND_JIT_TRACE_BLACKLISTED) {
```

Comment:
You may use ``jit_extension->func_info.flags`` check to stop tracing.
Just remove ``ZEND_FUNC_JIT_ON_HOT_TRACE`` flag.